File: apps/backend/app/schemas/workspace.py
from pydantic import BaseModel


class CreateWorkspaceRequest(BaseModel):
    name: str

    selected_repository: str | None = None

    workspace_question: str | None = None

    blueprint_target_file: str | None = None

    blueprint_target_symbol: str | None = None

    blueprint_target_reason: str | None = None

    selected_paper: str | None = None


class WorkspaceResponse(BaseModel):
    id: str

    name: str

    selected_repository: str | None

    workspace_question: str | None

    blueprint_target_file: str | None

    blueprint_target_symbol: str | None

    blueprint_target_reason: str | None

    selected_paper: str | None


class UpdateWorkspaceRequest(
    BaseModel,
):
    name: str | None = None

    selected_paper: str | None = None

    selected_repository: str | None = None

    workspace_question: str | None = None

    blueprint_target_file: str | None = None

    blueprint_target_symbol: str | None = None

    blueprint_target_reason: str | None = None

File: apps/frontend/types/workspace.ts
export interface Workspace {
  id: string;
  name: string;

  selected_repository?: string;
  workspace_question?: string;

  blueprint_target_file?: string;
  blueprint_target_symbol?: string;
  blueprint_target_reason?: string;

  selected_paper?: string;

  created_at?: string;
  updated_at?: string;
}

export interface CreateWorkspaceRequest {
  name: string;

  selected_repository?: string;
  workspace_question?: string;

  blueprint_target_file?: string;
  blueprint_target_symbol?: string;
  blueprint_target_reason?: string;

  selected_paper?: string;
}

export interface WorkspaceOutput {
  id: string;
  workspace_id: string;

  hybrid_result?: unknown;
  reproduction_result?: unknown;
  blueprint_result?: unknown;
  evaluation_result?: unknown;
}
